betterproto/tests/inputs/oneof_enum/test_oneof_enum.py

```diff
@@ -9,34 +9,36 @@
 from betterproto.tests.util import get_test_case_json_data
 
 
-@pytest.mark.xfail
 def test_which_one_of_returns_enum_with_default_value():
     """
     returns first field when it is enum and set with default value
     """
     message = Test()
     message.from_json(get_test_case_json_data("oneof_enum", "oneof_enum-enum-0.json"))
-    assert message.move is None
+
+    assert message.move == Move(
+        x=0, y=0
+    )  # Proto3 will default this as there is no null
     assert message.signal == Signal.PASS
     assert betterproto.which_one_of(message, "action") == ("signal", Signal.PASS)
 
```

Comment:
Is this change needed to solve the bug? As I understand the bug is in `to_dict` and not `from_dict`..

I think the idea that _proto doesn't have null values_ might actually be somewhat muddled, and incorrect. Because values can be skipped when they are sent over the wire, effectively making them undefined. However, the protobuf libraries will then simply return the default value for those skipped fields. That behavior is indeed part of protobuf specification, but for _oneof_ groups, I consider it not part of the spec that all other fields should have a default value as well (unless its actually wrtten in the spec, sorry I didn't have time to read it 😅 ). At least it doesn't make much sense to me why it would be needed.

Unless it is somehow needed for this bugfix, I would suggest to exclude that part from this PR, because its a functional change (and not sure if its an improvement). And that is a bit harder to judge than the bugfix, which we could merge quickly.

Reply:
So this is actually default behavior in Google's protobuf implementation as well. What's actually happening here is that the `move` field is never itself null but it isn't actually sent over the wire - calling to_dict() _should_ only show `signal` there. I think what's confusing is that a `oneof` field can have nothing set, an attribute set to a default value, OR an attrbiute set to a non-default value. But, on the actual message object, the attributes themselves are never actually null, they're always in a default state. I replicated this behaviour using these two messages:
```
message Point{
    int64 x = 1;
    int64 y = 2;
}

message Test{
    oneof test_value {
        Point point = 1;
        google.protobuf.BoolValue bool_value = 2;
        int64 int_value = 3;
    }
}
```

Compiling with the Google proto impl I then checked the following:
```python
In [4]: t = Test()

In [5]: t
Out[5]:

In [6]: t.point
Out[6]:

In [7]: t.point.x
Out[7]: 0
...

In [9]: type(t.point.x)
Out[9]: int

In [10]: type(t.point)
Out[10]: foo.Point

In [11]: type(t.int_value)
Out[11]: int

In [12]: t.bool_value
Out[12]:

In [13]: type(t.bool_value)
Out[13]: google.protobuf.wrappers_pb2.BoolValue

In [14]: from google.protobuf.json_format import MessageToDict

In [15]: MessageToDict(t)
Out[15]: {}

In [16]: t.point.x = 1

In [17]: MessageToDict(t)
Out[17]: {'point': {'x': '1'}}

In [18]: t.int_value = 1

In [19]: MessageToDict(t)
Out[19]: {'intValue': '1'}

In [20]: t.bool_value.value = True

In [21]: t
Out[21]:
bool_value {
  value: true
}

In [22]: MessageToDict(t)
Out[22]: {'boolValue': True}
```

I can either add an assert to this test case or a separate test case demonstrating that betterprotos has similar behaviour